### PR TITLE
layers: Implement vkCreateDescriptorPool() checks

### PR DESCRIPTION
**Pull request by krOoze**
Check explicit validity of `vkCreateDescriptorPool` command:
- `maxSets` > 0
- `pPoolSizes[].descriptorCount` > 0

\+ implement relevant tests

### CODE REVIEW

File: layers/parameter_validation_utils.cpp [hunk 1]
```diff
@@ -2344,6 +2344,35 @@ bool pv_vkDebugMarkerSetObjectNameEXT(VkDevice device, const VkDebugMarkerObject
...
+                if (pCreateInfo->pPoolSizes[i].descriptorCount <= 0) {
+                    skip |= log_msg(
+                        device_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_DESCRIPTOR_POOL_EXT,
+                        VK_NULL_HANDLE, __LINE__, VALIDATION_ERROR_04a0025c, LayerName,
+                        "vkCreateDescriptorPool(): pCreateInfo->pPoolSizes[%" PRIu32 "].descriptorCount is not greater than 0. %s",
+                        i, validation_error_map[VALIDATION_ERROR_04a0025c]);
+                }
+            }
+        }
+    }
```
Comment (tobine) [inline layers/parameter_validation_utils.cpp]: The "else" case here is VALIDATION_ERROR_1f611e01

**Comment by krOoze** [comment]
^ ups, forgot final Google Style uglyfication. Re-pushed.

File: layers/parameter_validation_utils.cpp (hunk 1 above)
Comment (krOoze) [reply layers/parameter_validation_utils.cpp]: Should be checked by the autogenerated part.
The purpose of the `if` here is only to protect against null dereference.

File: layers/parameter_validation_utils.cpp [hunk 2]
```diff
@@ -2378,7 +2407,7 @@ VKAPI_ATTR PFN_vkVoidFunction VKAPI_CALL vkGetPhysicalDeviceProcAddr(VkInstance
 
 // If additional validation is needed outside of the generated checks, a manual routine can be added to this file
 // and the address filled in here. The autogenerated source will call these routines if the pointers are not NULL.
-void InitializeManualParameterValidationFunctionPointers(void) {
+void InitializeManualParameterValidationFunctionPointers() {
```
Comment (krOoze) [inline layers/parameter_validation_utils.cpp]: Unrelated, but I can never resist killing random C-isms.

**Comment by mark-lunarg** [comment]
Thanks @krOoze!

**Pull request merged**